### DISCUSSION TITLE
Dialog cancel event handler

### DISCUSSION THREAD

**Discussion by qisbic** [fgilde/MudBlazor.Extensions#83] [Q&A] [answered]
The sample codes show detect when adding Alarm and ColorLens.. How do I detect a cancel/close?

 [JSInvokable]
    public void AlarmClick()
    {
        JsRuntime.InvokeVoidAsync("alert", "Alarm Clicked ");
    }
    [JSInvokable]
    public void ColorLensClick()
    {
        JsRuntime.InvokeVoidAsync("alert", "Color Lens Clicked");
    }

**Comment by fgilde** [comment, marked as answer]
In general `ShowEx<YourDialog>()` displays your dialog that means you internally have control about the result.

```c#
<MudDialog Class="mud-ex-dialog-initial">
    <TitleContent>
        <MudText Typo="Typo.h6">
            <MudIcon Icon="@icon" Class="mr-3"/> @title
        </MudText>
    </TitleContent>
    <DialogContent>
        Dialog Content
        <br/>
        @ContentMessage       
    </DialogContent>
    <DialogActions>
        <MudButton OnClick="Cancel">Cancel</MudButton>
        <MudButton Color="Color.Primary" OnClick="Submit">Ok</MudButton>
    </DialogActions>
</MudDialog>
@code {
    [CascadingParameter] MudDialogInstance MudDialog { get; set; }
    [Parameter] public string ContentMessage { get; set; }

    void Submit() => MudDialog.CloseAnimatedIf(DialogResult.Ok(true));
    void Cancel() => MudDialog.CancelAnimatedIf();
}
```

However from outside you just can access the result and check if it's canceled like this

```c#
IMudExDialogReference<SampleDialog>? dlgReference = await DialogService.ShowEx<SampleDialog>("Simple Dialog", dialog => { dialog.ContentMessage = "Hello from dialog sample page"; }, options);
var res = await dlgReference.Result;
if (res.Canceled)
{
    // canceled
}
```

**Reply by qisbic** [reply]
Hi,

What I am trying to do is intercept the "X" and pop up a message to ask if they want to close without saving.. I can't seem to do that.. with the example with AlarmClick.. you can do that.

![2024-06-24_12-12-24](https://github.com/fgilde/MudBlazor.Extensions/assets/17128669/f9bd8f63-5a78-4fe0-a0d7-4ca0042499c2)

**Reply by fgilde** [reply]
Ok understand. Can you create a feature request issue for that, then I will try add this , but however you can also currently do this. Just set closable in the dialogOptions to false and add an own custom close button like the test alarm button, but i will try to implement an cancable event in the ÌDialogEventService`

**Comment by fgilde** [comment]
Its done in 1.7.89

Demo:
https://www.mudex.org/dialogs-closing-event
